server/application/application.go

```diff
@@ -1113,15 +1113,26 @@ func (s *Server) validateAndNormalizeApp(ctx context.Context, app *appv1.Applica
 			return err
```

Comment:
I think we need to update sync API to support multiple sources. Otherwise syncing from UI won't work which is a blocker.